### PR TITLE
mpfr: Fix URL

### PR DESCRIPTION
**Pull request by jeremy-coulon**
Only the latest version is available at https://www.mpfr.org/mpfr-current

Specify library name and version:  **mpfr/4.0.2**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Why we need to change the URL, whether the official homepage points to the current url? 

https://www.mpfr.org/mpfr-current/#download

**Comment by jeremy-coulon** [comment]
I need to rebuild this package from source. It fails because previous URL ('mpfr-current') is not reliable.

File: recipes/mpfr/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 sources:
   "4.0.2":
     sha256: ae26cace63a498f07047a784cd3b0e4d010b44d2b193bab82af693de57a19a78
-    url: https://www.mpfr.org/mpfr-current/mpfr-4.0.2.tar.gz
+    url: https://ftp.gnu.org/gnu/mpfr/mpfr-4.0.2.tar.gz
```
Comment (uilianries) [inline recipes/mpfr/all/conandata.yml:4]:
```suggestion
    url: [
      "https://www.mpfr.org/mpfr-current/mpfr-4.0.2.tar.gz",
      "https://ftp.gnu.org/gnu/mpfr/mpfr-4.0.2.tar.gz",
    ]
```

Use gnu url as a mirror instead. I can download from mpfr, so probably it was a temporary error.

Comment (jeremy-coulon) [reply recipes/mpfr/all/conandata.yml:4]: Are you sure?
```
$ wget https://www.mpfr.org/mpfr-current/mpfr-4.0.2.tar.gz
--2020-07-24 14:53:48--  https://www.mpfr.org/mpfr-current/mpfr-4.0.2.tar.gz
Resolving www.mpfr.org (www.mpfr.org)... 152.81.144.153
Connecting to www.mpfr.org (www.mpfr.org)|152.81.144.153|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2020-07-24 14:53:48 ERROR 404: Not Found.
```

Comment (uilianries) [reply recipes/mpfr/all/conandata.yml:4]: My bad, I downloaded https://www.mpfr.org/mpfr-current/mpfr-4.1.0.tar.gz, the latest version available.

**Pull request merged**